pkg/csi/service/vanilla/controller_test.go

```diff
@@ -400,7 +419,79 @@ func TestCreateVolumeWithStoragePolicy(t *testing.T) {
 		t.Fatalf("Volume should not exist after deletion with ID: %s", volID)
 	}
 }
+*/
+func TestCreateVolumeWithMultipleDatastores(t *testing.T) {
```

Comment:
Can you explain a bit more on which datastore is ignored and which is not? I did not understand what this test is doing.

Reply:
This function will be executed by unit test and integrate test.
When running in unit test,  GetDatastoreIgnoreMapForBlockVolumes will return empty, and no datastores will be filtered out.
When running in integrate test, GetDatastoreIgnoreMapForBlockVolumes will scan all datastores on the real  VC and compute this map.

Reply:
>When running in unit test, GetDatastoreIgnoreMapForBlockVolumes will return empty, and no datastores will be filtered out.

This is not really unit testing the functionality you are adding. Can you add an unit test where some datastores are getting ignored? If this takes time, you can file an issue to add an unit test and create a follow on change.